src/main/scala/is/hail/utils/ArrayBuilder.scala

```diff
@@ -36,6 +36,28 @@ class ArrayBuilder[@specialized T](initialCapacity: Int)(implicit tct: ClassTag[
     size_ += 1
   }
 
+  def +(x: T): ArrayBuilder[T] = {
```

Comment:
builders are not meant to be used as functional collections. If we change the aggregate to groupByKey, this stuff can go away.